developer-guide.md

```diff
@@ -0,0 +1,184 @@
+mvn package -DskipTests
+mvn install -DskipTests
+```
+
+## Project setup using an IDE
+
+If you are using an IDE like [IntelliJ IDEA](https://www.jetbrains.com/idea/) or [Eclipse](https://eclipseide.org/)
+to help with your code development you should set up source and target folders so that the IDE's compiler can find all
+the source code including the generated sources (after running `mvn install -DskipTests`).
+
+For IntelliJ this can be done by going to **File > Project Structure ... > Modules**:
+
+- **Source Folders**
+    - src/main/java
+    - src/main/proto
+    - target/generated-sources/protobuf/grpc-java (generated)
+    - target/generated-sources/protobuf/java (generated)
+- **Test Source Folders**
+    - src/test/java
+    - target/generated-test-sources/protobuf/grpc-java (generated)
+    - target/generated-test-sources/protobuf/java (generated)
+- **Resource Folders**
+    - src/main/resources
+- **Test Resource Folders**
+    - src/test/resources
+- **Excluded Folders**
+    - target
+
+You may also want to increase your Java Heap size to at least 1.5 GB.
+
+## Testing code changes
+
+Note, before running the test cases, make sure you have `etcd` installed:
+
+```Bash
+$ etcd --version
+
+etcd Version: 3.5.5
+Git SHA: 19002cfc6
+Go Version: go1.19.1
+Go OS/Arch: darwin/amd64
+```
+
+On `macOS` you can install `etcd` with [Homebrew](https://brew.sh/):
+
+```Bash
+brew install etcd
+```
+
+You can either run all test suites at once. You can use the `-q` flag to reduce noise:
+
+```Bash
+mvn test -q
+```
+
+Or you can run individual test cases:
+
+```Bash
+mvn test -Dtest=ModelMeshErrorPropagationTest
+mvn test -Dtest=SidecarModelMeshTest,ModelMeshFailureExpiryTest
+```
+
+It can be handy to use `grep` to reduce output noise:
+
+```Bash
+mvn test -Dtest=SidecarModelMeshTest,ModelMeshFailureExpiryTest | \
+  grep -E " Running |\[ERROR\]|Failures|SUCCESS|Skipp|Total time|Finished"
+
+[INFO] Running com.ibm.watson.modelmesh.ModelMeshFailureExpiryTest
+[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 10.257 s - in com.ibm.watson.modelmesh.ModelMeshFailureExpiryTest
+[INFO] Running com.ibm.watson.modelmesh.SidecarModelMeshTest
+[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 17.302 s - in com.ibm.watson.modelmesh.SidecarModelMeshTest
+[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0
+[INFO] BUILD SUCCESS
+[INFO] Total time:  39.916 s
+[INFO] Finished at: 2022-11-01T14:33:33-07:00
+```
+
+## Building the container image
+
+Sample build:
+
+```bash
+export DOCKER_USER=$(
+  docker-credential-$(jq -r .credsStore ~/.docker/config.json) list | \
+  jq -r '. | to_entries[] | select(.key | contains("docker.io")) | last(.value)' | head -1
+)
+export IMAGE_TAG="dev" # "v0.10.0", "latest"
+export IMAGE_NAME="${DOCKER_USER}/modelmesh"
+export GIT_COMMIT=$(git rev-parse HEAD)
+export BUILD_ID=$(date '+%Y%m%d')-$(git rev-parse HEAD | cut -c -5)
+
+docker build -t ${IMAGE_NAME}:${IMAGE_TAG} \
```

Comment:
This section is useful with all the scripts, may be a bit error prone too - esp. a person is not so shell scripting savvy! 

Can it be summarised, by saying `for example`:

 Just 2 Steps.

1. `docker build -t yourdockerusername/modelmesh:dev`
2. `docker push yourdockerusername/modelmesh:dev`

Reply:
The Docker build needs the build arguments:

```Dockerfile
ARG imageVersion
ARG buildId
ARG commitSha
```